torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1942,6 +2006,35 @@ def _should_allocate(user_marked_dynamic, assume_static_by_default):
 def _is_dim_dynamic(t, d):
     return hasattr(t, "_dynamo_dynamic_indices") and d in t._dynamo_dynamic_indices
 
+class MinMaxConstraint:
```

Comment:
I don't really understand why we need another class for this

Reply:
This seems overly specialized for min max constraint. Didn't we agree that there were more than one potential constraint type? Better just have a bucket constraint data type that has room to grow rather than call it MinMaxConstraint.

Reply:
> Didn't we agree that there were more than one potential constraint type?

I thought we explicitly said we only have min max constrains?

Reply:
min max to start, see https://docs.google.com/document/d/1aoIyYE8_6cYpWqS25thzVoIiKsT5aaUEOiiPwbIXt8k/edit?disco=AAAAsiJ5ujg

Reply:
I thought we disagreed on that, and said we would punt it until we find a need. If this is indeed the case, I am happy to refactor.

Reply:
I think in the March 7th 11:45 PT discussion, we agreed on expressing min-max only and only add things when they are proven to be useful. 

In the referenced document, it says we may "add more stuff", what are included the "more stuff"?